Swap API call for passwordless signup to use usersNew endpoint

andrewserong opened the pull request:
#### Changes proposed in this Pull Request

* Move passwordless signup to use the `usersNew` method with an `is_passwordless` flag
* Also pass in current flowName and A/B test variations

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Apply D35282-code
* Go to `http://calypso.localhost:3000/start` and select the `passwordless` variant of the `passwordlessSignup` A/B test
* Enter an existing email address, and you should see the same error notification as before, asking you to log in
* Create a new account, and complete signup

p-jackson commented on client/blocks/signup-form/passwordless.jsx:
```diff
@@ -153,11 +158,8 @@ class PasswordlessSignupForm extends Component {
 				);
 			default:
-				return (
-					errorObj.message ||
```
I was going to question why you removed the default of showing the error message from the server. But after taking a look at the code I see that the error messages returned by the server aren't exactly user friendly.

andrewserong replied:
Yep, that's the only reason I removed it!

andrewserong commented:
Thanks for reviewing @p-jackson!

> Not related to your change, just a potential bit of refactor: why don't we make onFormSubmit async so we can tidy up the promise machinery?

That's a good idea. I'll take a look at cleaning this up today.

andrewserong commented:
@p-jackson I've converted `onFormSubmit` to an async method and renamed the callback to `createAccountCallback` to match the naming of the `createAccount` function in `step-actions` that this call to `usersNew` kind of matches. What do you think?
